parser.y

```diff
@@ -406,6 +406,13 @@ decl: declaration_specifiers init_declarator_list_opt SEMICOLON {
     } else {
       auto decl = std::move(std::get<std::unique_ptr<DeclNode>>(decl_specifiers));
       auto* rec_decl = dynamic_cast<RecordDeclNode*>(decl.get());
+      // NOTE: This is record type declaration. If there are objects in init_decl_list,
+      // then it should be record variable declaration.
+      if (init_decl_list.empty()) {
+        decl_list.push_back(std::move(decl));
+      }
```

Comment:
Suggest placing this piece of code after the declaration of `rec_decl`, as `rec_decl` is not used inside it.